client/src/components/EditExperiment.jsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState } from "react";
+import { editExperiment } from "../actions/exptActions";
+import { useDispatch } from "react-redux";
+
+
+const EditExperiment = ({ id, name, description, duration, date_started, date_ended, setIsEditing }) => {
+  const dispatch = useDispatch();
+  const [newName, setNewName] = useState(name);
+  const [newDescription, setNewDescription] = useState(description);
+  const [newDuration, setNewDuration] = useState(duration);
+  const inputCSS = "border border-primary-oxfordblue rounded-lg px-2";
+
+  const startDate = new Date(date_started).toLocaleDateString("en-US");
+
+  const getChangedData = () => {
+    const edits = {}
+    if (newName !== name) edits.name = newName;
+    if (Number(newDuration) !== Number(duration)) edits.duration = newDuration;
+    if (newDescription !== description) edits.description = newDescription;
+    return edits;
+  }
+
+  const validateInput = () => {
+    let errMessage = "";
+
+    if (newDuration < 1) {
+      errMessage += "- The duration has to be at least one day\n"
+    }
+
+    if (description.length > 255) {
+      errMessage += "- The length of the description is too long\n"
+    }
+
+    if (name.length > 50) {
```

Comment:
Should these be `newDescription` and `newName`?

Reply:
Yeah, just updated them. Good catch!